src/main.rs

```diff
@@ -0,0 +1,227 @@
+use calamine::{open_workbook_auto,  Reader};
+use chrono::{Datelike, NaiveDate, Utc};
+use clap::Parser;
+use std::fs::File;
+use std::io::{BufRead, Write};
+use std::collections::HashMap;
+
+#[derive(Parser)]
+struct Args {
+    #[arg(short = 'e', long = "emp-data-file-path")]
+    emp_data_file_path: String,
+
+    #[arg(short = 'd', long = "dept-data-file-path")]
+    dept_data_file_path: String,
+
+    #[arg(short = 's', long = "salary-data-file-path")]
+    salary_data_file_path: String,
+
+    #[arg(short = 'l', long = "leave-data-file-path")]
+    leave_data_file_path: String,
+
+    #[arg(short = 'o', long = "output-file-path")]
+    output_file_path: String,
+}
+
+struct Employee {
+    emp_id: i32,
+    emp_name: String,
+    dept_id: i32,
+    mobile_no: String,
+    email: String,
+}
+
+ #[derive(Debug)]
+struct Department {
+    #[allow(dead_code)]
+    dept_id: i32,
+    dept_title: String,
+}
+
+struct Salary {
+    emp_id: i32,
+    salary_date: String,
+    salary_status: String,
+}
+
+struct Leave {
+    emp_id: i32,
+    leave_from: NaiveDate,
+    leave_to: NaiveDate,
+}
+
+// Load Employee Data
+fn load_employee_data(file_path: &str) -> Vec<Employee> {
+    let file = File::open(file_path).expect("Could not open employee data file");
+    let reader = std::io::BufReader::new(file);
+    
+    reader.lines().skip(1).filter_map(|line| {
+        let line = line.ok()?;
```

Comment:
Use expect instead of ? to handle errors better. Expect allows to explain the exception.